test/apm_ec2/apm_test.go

```diff
@@ -0,0 +1,147 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build !windows
+
+package apm_ec2
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/environment"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric/dimension"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/status"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/test_runner"
+)
+
+const testRetryCount = 3
+const namespace = "AWS/APM"
+
+var path, testId string
+
+func init() {
+	environment.RegisterEnvironmentMetaDataFlags()
+
+	flag.StringVar(&path, "path", "./resources/run_java_application.sh", "path to java script file.")
+	flag.StringVar(&testId, "test.id", "unknown", "testing id.")
+}
+
+type APMEC2TestRunner struct {
+	test_runner.BaseTestRunner
+}
+
+func (runner *APMEC2TestRunner) SetupAfterAgentRun() error {
```

Comment:
I'm assuming this is just checking that the sample app is working? Maybe needs a more clear function name like `SetupSampleJavaApplication()` or something